goldmane/pkg/emitter/http_client.go

```diff
@@ -80,7 +81,19 @@ func newEmitterClient(url, caCert, clientKey, clientCert, serverName string) (*e
 		return nil, err
 	}
 
-	updChan := make(chan struct{})
+	// We only reload the client when we need to use it, but the file watcher may generate events at any time - sometimes
+	// several in quick succession. To handle this, we use a small deduplicator goroutine to ensure we're servicing the
+	// channel.
+	updChan := make(chan struct{}, 1)
+	reloadClient := make(chan struct{})
+	go func() {
```

Comment:
Instead of putting this here, what would you think of the file watcher just keeping a hash of file and only sending notifications if the contents has changed?

I think this would be a bit more robust and require less from any of you're code that uses it.

This current implementation doesn't really help deduplicate if the reader side of the channel reads the struct off before the next update happens anyway.

Reply:
The way that configmap hot reloading works involves multiple temporary files with randomly generated names that all get copied around / deleted / updated. It results in a few successive fsnotify events of varying types, which get de-duplicated here.

We could have the file-watcher try to do that itself by using memory to cache things, but it will be pretty awkward.

If anything, the hash checking should be done on this side of the channel, where we know exactly which files we care about (the file watcher watches a directory by necessity). 

But even that I think is overkill. In normal operation this is something that will happen once, and there's not really a downside to us getting multiple events here. The upside is less code.